### PR TITLE
Bluetooth: has: Handle active preset selection

### PR DESCRIPTION
**Pull request by MariuszSkamra**
This adds handling of active preset selection in HAS.

### CODE REVIEW

File: include/zephyr/bluetooth/audio/has.h [hunk 1]
```diff
@@ -131,6 +131,26 @@ int bt_has_client_discover(struct bt_conn *conn);
  */
 int bt_has_client_conn_get(const struct bt_has *has, struct bt_conn **conn);
 
+/** @brief Preset operations structure. */
+struct bt_has_preset_ops {
```
Comment (Thalley) [inline include/zephyr/bluetooth/audio/has.h:135]: Are you expecting to add more callbacks to this struct?

File: subsys/bluetooth/audio/has.c [hunk 2]
```diff
@@ -514,12 +678,22 @@ int bt_has_preset_register(const struct bt_has_preset_register_param *param)
 		BT_WARN("param->name is too long (%zu > %u)", name_len, BT_HAS_PRESET_NAME_MAX);
 	}
 
+	CHECKIF(param->ops == NULL) {
+		BT_ERR("param->ops is NULL");
+		return -EINVAL;
+	}
+
+	CHECKIF(param->ops->select == NULL) {
+		BT_ERR("param->ops->select is NULL");
```
Comment (Thalley) [inline subsys/bluetooth/audio/has.c:687]: I prefer using `BT_DBG` instead of `BT_ERR` when we are already returning an error code to the caller

**Review by Thalley — DISMISSED** [review]
Minor comments that can be handled, but otherwise LGTM

File: include/zephyr/bluetooth/audio/has.h (hunk 1 above)
Comment (MariuszSkamra) [reply include/zephyr/bluetooth/audio/has.h:135]: Yes, at least one

**Comment by MariuszSkamra** [comment]
rebased

File: tests/bluetooth/bsim_bt/bsim_test_audio/src/has_test.c [hunk 3]
```diff
@@ -18,6 +18,15 @@ const char *test_preset_name_1 = "test_preset_name_1";
...
 const enum bt_has_properties test_preset_properties = BT_HAS_PROP_AVAILABLE;
 
+static int preset_select(uint8_t index, bool sync)
+{
+	return 0;
+}
+
+static const struct bt_has_preset_ops preset_ops = {
+	.select = preset_select,
+};
```
Comment (Thalley) [inline tests/bluetooth/bsim_bt/bsim_test_audio/src/has_test.c:28]: Is the client selecting a preset? Should we verify that?

Comment (MariuszSkamra) [reply tests/bluetooth/bsim_bt/bsim_test_audio/src/has_test.c:28]: I will extend the tests in suceeding PR

**Review by Thalley — APPROVED** [review]

**Review by sjanc — APPROVED** [review]

**Pull request merged**